website/addons/evernote/model.py

```diff
@@ -0,0 +1,128 @@
+
+from website.addons.base import AddonOAuthUserSettingsBase, AddonOAuthNodeSettingsBase
+from website.addons.base import StorageAddonBase
+from website.addons.evernote import (settings, utils)
+from website.addons.evernote.serializer import EvernoteSerializer
+from website.oauth.models import (ExternalProvider, OAUTH1)
+
+from modularodm import fields
+
+import logging
+logger = logging.getLogger(__name__)
+
+class Evernote(ExternalProvider):
+    """
+    First cut at the Evernote provider
+
+    """
+    name = 'Evernote'
+    short_name = 'evernote'
+
+    # Evernote has a sandbox and a production service
+    # with different endpoints
+
+    client_id = settings.EVERNOTE_CLIENT_ID
+    client_secret = settings.EVERNOTE_CLIENT_SECRET
+    _sandbox = settings.EVERNOTE_SANDBOX
+
+    BASE_URL = "https://www.evernote.com" if not _sandbox  \
+           else "https://sandbox.evernote.com"
+
+    auth_url_base = '{}/OAuth.action'.format(BASE_URL)
+    request_token_url = '{}/oauth'.format(BASE_URL)
+    callback_url = '{}/oauth'.format(BASE_URL)
+
+    _oauth_version = OAUTH1
+
+    def handle_callback(self, response):
+        """View called when the Oauth flow is completed. Adds a new BoxUserSettings
+        record to the user and saves the user's access token and account info.
+        """
+
+        client = utils.get_evernote_client(token=response.get('oauth_token'))
+
+        userStore = client.get_user_store()
+        user = userStore.getUser()
+
+        return {
+            'provider_id': user.id,  # or user.username
+            'display_name': user.name,
+            'profile_url': ''
+        }
+
+class EvernoteUserSettings(AddonOAuthUserSettingsBase):
+    oauth_provider = Evernote
+    serializer = EvernoteSerializer
+
+
+class EvernoteNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
```

Comment:
This class will need 

* an `api` property that returns a possibly authenticated `Evernote(ExternalProvider)` session
* a `deauthorize` method that logs the action and calls `clear_settings` (also required addition) and `clear_auth` (`super` version is fine)
* a `clear_settings` method that unsets 'folder' related fields
* a `serialize_waterbutler_credentials` method that prepares auth credentials to be passed to WB
* a `serialize_waterbutler_settings` method that tells WB what folder (Notebook) to look for files (Notes) in

It may also require a `create_waterbutler_log` eventually -- when write functionality is added -- but it shouldn't need it now.